cmake/modules/CUDA.cmake

```diff
@@ -199,6 +202,7 @@ function(dgl_select_nvcc_arch_flags out_variable)
   string(REPLACE ";" " " __nvcc_archs_readable "${__nvcc_archs_readable}")
   set(${out_variable}          ${__nvcc_flags}          PARENT_SCOPE)
   set(${out_variable}_readable ${__nvcc_archs_readable} PARENT_SCOPE)
+  set(CUDA_ARCHITECTURES       ${__archs}               PARENT_SCOPE)
```

Comment:
what's the relationship between `CUDA_ARCHITECTURES` and existing `CUDA_ARCH_NAME`: https://github.com/dmlc/dgl/blob/3cf8246291d2ab559caae7f3bfafcb25a436405b/cmake/modules/CUDA.cmake#L124C8-L124C22. For now, we're using `CUDA_ARCH_NAME` alongside with `CUDA_ARCH_BIN` and `TORCH_CUDA_ARCH_LIST` to compile cuda/ptx for different archs.

Reply:
`CUDA_ARCH_NAME` is a name like `Auto`, `All`, or a specific architecture like `Kepler`. However, specifying this option for DGL doesn't affect Graphbolt at all. Thus, I extract the SM version such as 50 or 80 etc. into `CUDA_ARCHITECTURES` and pass it to Graphbolt via `CUDAARCHS`. Then, setting `CMAKE_CUDA_ARCHITECTURES` for Graphbolt enables the same targets as DGL.

Reply:
could you check with `libdgl.so` and `dgl/dgl_sparse/libdgl_sparse_pytorchxxx` as well?

Reply:
dgl_sparse doesn't have any `.cu` files at all in its src directory.

Reply:
`libdgl.so` has all architectures when `All` option is passed. And:
```
mfbalin@BALIN-PC:~/dgl-1$ cuobjdump /home/mfbalin/dgl-1/build/dgl_sparse/libdgl_sparse_pytorch_2.0.1.so -sass -ptx | grep "code for sm_"
cuobjdump info    : File '/home/mfbalin/dgl-1/build/dgl_sparse/libdgl_sparse_pytorch_2.0.1.so' does not contain device code
```